prov/shm/src/smr_av.c

```diff
@@ -148,8 +152,17 @@ static int smr_av_insert(struct fid_av *av_fid, const void *addr, size_t count,
 		if (ret) {
 			if (fi_addr)
 				fi_addr[i] = util_addr;
-			if (shm_id >= 0)
-				smr_map_del(&smr_av->smr_map, shm_id);
+			if (shm_id >= 0) {
+				node = ofi_rbmap_find(&smr_av->smr_map.rbmap,
+				      &smr_av->smr_map.peers[shm_id].peer.name);
+				assert(node);
+				ret = smr_map_del(&smr_av->smr_map.rbmap,
+						  node, NULL);
```

Comment:
Have you considered the possibility that `node` may become invalid when `smr_map_del` is called?